include/micm/solver/backward_euler.inl

```diff
@@ -0,0 +1,204 @@
+/* Copyright (C) 2023-2024 National Center for Atmospheric Research
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+enum class MicmBackwardEulerErrc
+{
+  FailedToConverge = 1
+};
+
+namespace std
+{
+  template<>
+  struct is_error_condition_enum<MicmBackwardEulerErrc> : true_type
+  {
+  };
+}  // namespace std
+
+namespace
+{
+  class BackwardEulerErrorCategory : public std::error_category
+  {
+   public:
+    const char* name() const noexcept override
+    {
+      return "MICM BackwardEuler";
+    }
+    std::string message(int ev) const override
+    {
+      switch (static_cast<MicmBackwardEulerErrc>(ev))
+      {
+        case MicmBackwardEulerErrc::FailedToConverge:
+          return "Failed to converge";
+        default: return "Unknown error";
+      }
+    }
+  };
+
+  const BackwardEulerErrorCategory backwardEulerErrorCategory{};
+}  // namespace
+
+inline std::error_code make_error_code(MicmBackwardEulerErrc e)
+{
+  return { static_cast<int>(e), backwardEulerErrorCategory };
+}
+
+namespace micm
+{
+  inline BackwardEuler::BackwardEuler()
+  {
+  }
+
+  inline void BackwardEuler::Solve(double time_step, auto& state, auto linear_solver, auto process_set, const std::vector<micm::Process>& processes, auto jacobian_diagonal_elements)
+  {
+    // A fully implicit euler implementation is given by the following equation:
+    // y_{n+1} = y_n + H * f(t_{n+1}, y_{n+1})
+    // This is a root finding problem because you need to know y_{n+1} to compute f(t_{n+1}, y_{n+1})
+    // you need to solve the equation y_{n+1} - y_n - H f(t_{n+1}, y_{n+1}) = 0
+    // We will also use the same logic used by cam-chem to determine the time step
+    // That scheme is this: 
+    // Start with H = time_step
+    // if that fails, try H = H/2 several time
+    // if that fails, try H = H/10 once
+    // if that fails, accept the current integration and move on. Continue on with the current H until the end of the time step
```

Comment:
Since I forgot this all the time, can you please add a sentence saying **We accept the current H but do not update the Yn vector**?

Reply:
done